src/metldata/builtin_transformations/count_content_values/data_transform.py

```diff
@@ -36,9 +37,9 @@
 )
 
 
-def get_class_resources(*, data: DataPack, class_name: str):
-    """Extract the resources of a given class from a Datapack."""
-    resources = data.resources.get(class_name)
+def get_class_resources(*, data: dict, class_name: str) -> dict:
+    """Extract the resources of a given class from the dictionary."""
+    resources = data.get("resources", {}).get(class_name)
     if not resources:
         raise EvitableTransformationError()
```

Comment:
The rest of the code indexes using a key because those functions receive a datapack as input and handle its conversion to a dictionary within the same function. Thus, these functions assume the required keys are present in the converted data—otherwise, the datapack validation would fail before the conversion even occurs. This is the key distinction between those functions and this one.

In this case, if the key "resources" is missing in the data, it will still result in an [EvitableTransformationError](https://github.com/ghga-de/metldata/blob/ab9093764a6ad99ac7553c4065e26c9cea15c7e7/src/metldata/transform/exceptions.py#L57), as it would have already been caught earlier in the `count_content` function. Consequently, the program wouldn't even reach this function. In my view, the purpose of EvitableTransformationError is not to be overly descriptive—it’s a fallback mechanism. That is why I don't agree that we have to differentiate the cause between two possible problems in this function.